lib/Yield/Support/SugiyamaStyleGraphLayout/HorizontalPositions.cpp

```diff
@@ -193,3 +194,76 @@ void setHorizontalCoordinates(const LayerContainer &Layers,
     }
   }
 }
+
+void setStaticOffsetHorizontalCoordinates(const LayerContainer &Layers,
```

Comment:
This function is pretty dense, and there isn't a single line of comment. Can you please add comments to guide the reader through the reasoning behind this function?

Reply:
Does this work? https://github.com/revng/revng/pull/191/commits/2d534d73cd7bddc11ccf1a451b7a0757833c5db9

Or do some part need more detailed explanation?

Reply:
This is great! Thanks a lot! I wish all the code I read was commented like this. Take it as a reference for all the future code you write!

Reply:
BTW, the `preservingSliceImpl` would really benefit from this level of documentation too.

Reply:
On it!

Reply:
Not on the same level (that function is not as dense), but should be helpful nevertheless: [`caf19f9`](https://github.com/revng/revng/pull/191/commits/caf19f96a318e5457c808c0a1bee2a874918edba)

Reply:
Great, thanks!